packages/trpc/server/routers/loggedInViewer/addNotificationsSubscription.handler.ts

```diff
@@ -10,20 +15,52 @@ type AddSecondaryEmailOptions = {
   input: TAddNotificationsSubscriptionInputSchema;
 };
 
+const log = logger.getSubLogger({ prefix: ["[addNotificationsSubscriptionHandler]"] });
+
 export const addNotificationsSubscriptionHandler = async ({ ctx, input }: AddSecondaryEmailOptions) => {
   const { user } = ctx;
   const { subscription } = input;
 
+  const parsedSubscription = subscriptionSchema.safeParse(JSON.parse(subscription));
+
+  if (!parsedSubscription.success) {
+    log.error("Invalid subscription", parsedSubscription.error, JSON.stringify(subscription));
+    throw new TRPCError({
+      code: "BAD_REQUEST",
+      message: "Invalid subscription",
+    });
+  }
+
   const existingSubscription = await prisma.notificationsSubscriptions.findFirst({
-    where: { userId: user.id, subscription },
+    where: { userId: user.id },
   });
 
   if (!existingSubscription) {
     await prisma.notificationsSubscriptions.create({
       data: { userId: user.id, subscription },
     });
+  } else {
+    await prisma.notificationsSubscriptions.update({
+      where: { id: existingSubscription.id },
+      data: { userId: user.id, subscription },
+    });
```

Comment:
Does this mean a user can only have single subscription? 🤔

Reply:
yeah right now just one.